### ISSUE TITLE
Import modules unresolved

### ISSUE THREAD

**Issue by MarlonSchultz** [klazuka/intellij-elm#559]
## Problem

A local module is marked as unused (not installed via elm). The function call is marked as unresolved reference. 

## What have I tried this far?

- elm.json is connected and loads correctly (altho the module is not mentioned in this file since its not installed via elm install)
- external libraries are shown
- elm-stuff is included in intellij
- cache has been invalidated

### Url.elm

``` elm
module Url exposing (getUrl)


getUrl : String
getUrl =
    "http://localhost:8080/"
```

### NewType.elm referencing Url.elm

``` elm
module NewType exposing (init, main)

import Browser
import Game exposing (Msg)
import Html exposing (Html, div, text)
import Url exposing (getUrl) -- marked as not used


main =
    Browser.element
        { init = init
        , update = update
        , subscriptions = always Sub.none
        , view = view
        }


type alias Model =
    Int


init : () -> ( Model, Cmd Msg )
init _ =
    ( 1, Cmd.none )


useImport : String
useImport =
    getUrl  -- marked as unresolved


update : Msg -> Model -> ( Model, Cmd Msg )
update msg model =
    ( model, Cmd.none )


view : Model -> Html Msg
view model =
    div [] [ text "something" ]
```

![image](https://user-images.githubusercontent.com/14779331/69901926-0941ab00-1388-11ea-90de-1941802f179c.png)

![image](https://user-images.githubusercontent.com/14779331/69901945-532a9100-1388-11ea-8749-cda59a066539.png)

**Comment by MalmarPadecain** [comment]
Running Webstorm 2019.3 with plugin version 3.4.3 I get a different Error message: `Unresolved reference 'getUrl'`. 
The code compiles without issues. It seems like the plugin is not aware that shadowing imports is a thing.

**Comment by klazuka** [comment]
We do try to handle shadowing correctly, but I never thought about this case. The problem is that there are 2 `Url` modules visible: one defined in your project and the other which is an indirect dependency. I assumed that you could never have 2 modules with the same name in the same project, but I forgot about the indirect dependency case. I'll see what I can do to fix it.

In the meantime, you can workaround the issue by giving your Url module a different name.

**Comment by klazuka** [comment]
I have to ship a fix for IntelliJ 2019.3, so I don't have time to fix this right now. But making a note for myself in the future: the bad reference is from `ElmImportClause` where it uses `ElmModulesIndex.get` to lookup the module by name. That function internally uses `firstOrNull` as a crude way to resolve ambiguity. But we need to do something smarter here to decide in which cases modules belonging to *indirect dependencies*  should (or should not) be allowed.

**Comment by MarlonSchultz** [comment]
## Temporary fix (for ppl browsing issues)

Renaming the module Url to MyUrl.elm works as mentioned beforehand.

**Comment by ajalt** [comment]
This should be fixed by #614. If the issue still exists in the latest version of the plugin (3.5.1), please open a new issue.
